contracts/modules/UpgradableModuleFactory.sol

```diff
@@ -0,0 +1,88 @@
+pragma solidity ^0.5.0;
+
+import "./ModuleFactory.sol";
+import "../proxy/OwnedUpgradeabilityProxy.sol";
+
+/**
+ * @title Factory for deploying upgradable modules
+ */
+contract UpgradableModuleFactory is ModuleFactory {
+
+    event LogicContractSet(string _version, address _logicContract, bytes _logicData);
+
+    struct LogicContract {
+        string version;
+        address logicContract;
+        bytes logicData;
+    }
+
+    // Mapping from version to logic contract
+    mapping (uint256 => LogicContract) logicContracts;
+
+    // Mapping from Security Token address to module version
+    mapping (address => uint256) moduleVersions;
+
+    // Mapping from Security Token address to module address
+    mapping (address => address) modules;
```

Comment:
This will mess up when the issuer adds multiple of same module (multiple sto etc)

Reply:
Good point... Will fix this.

Reply:
@maxsam4 Fixed - please review.